pkg/operation/botanist/component/dependencywatchdog/bootstrap.go

```diff
@@ -362,17 +197,180 @@ func (b *bootstrapper) WaitCleanup(ctx context.Context) error {
 	return managedresources.WaitUntilDeleted(timeoutCtx, b.client, b.namespace, b.name())
 }
 
+func (b *bootstrapper) deleteOldDWDManagedResources(ctx context.Context) error {
+	log := logf.FromContext(ctx)
+	log.Info("Cleaning up old out of support managed resources for DWD")
+	if err := managedresources.DeleteForSeed(ctx, b.client, b.namespace, ManagedResourceDependencyWatchdogProbe); err != nil && !apierrors.IsNotFound(err) {
+		return fmt.Errorf("could not delete managed resource %s: %w", ManagedResourceDependencyWatchdogProbe, err)
+	}
+	if err := b.waitUntilManagedResourceDeleted(ctx, ManagedResourceDependencyWatchdogProbe); err != nil {
+		return err
+	}
+
+	if err := managedresources.DeleteForSeed(ctx, b.client, b.namespace, ManagedResourceDependencyWatchdogEndpoint); err != nil && !apierrors.IsNotFound(err) {
+		return fmt.Errorf("could not delete managed resource %s: %w", ManagedResourceDependencyWatchdogEndpoint, err)
+	}
+	if err := b.waitUntilManagedResourceDeleted(ctx, ManagedResourceDependencyWatchdogEndpoint); err != nil {
+		return err
+	}
+	return nil
+}
+
+func (b *bootstrapper) waitUntilManagedResourceDeleted(ctx context.Context, name string) error {
+	timeoutCtx, cancel := context.WithTimeout(ctx, TimeoutWaitForManagedResource)
+	defer cancel()
+	if err := managedresources.WaitUntilDeleted(timeoutCtx, b.client, b.namespace, name); err != nil {
+		return fmt.Errorf("error while waiting for managed resource %s to be deleted: %w", name, err)
+	}
+	return nil
+}
+
+func (b *bootstrapper) getConfigMap() (*corev1.ConfigMap, error) {
+	var (
+		config string
+		err    error
+	)
+
+	switch b.values.Role {
+	case RoleWeeder:
+		config, err = encodeConfig[weederapi.Config](&b.values.WeederConfig)
```

Comment:
The type is not needed when calling a generic function.
```suggestion
		config, err = encodeConfig(&b.values.WeederConfig)
```